src/artrace.py

```diff
@@ -934,412 +978,72 @@ def model_tilt_test(slf, det, msarc, guesstilts=None, censpec=None, plotQA=False
-                    tilts = tiltval[:,np.newaxis].repeat(tiltang.shape[1],axis=1)
-                else:
-                    msgs.warn("There are still not enough detections to obtain a reliable tilt trace")
-                    msgs.info("Assuming there is no tilt")
-                    tilts = np.zeros_like(slf._lordloc)
-        if guesstilts is not None:
-            tilts += guesstilts
-    else:
-        msgs.error("Please use trace+tilts+fit1d")
+            wmask = np.where(mtfit == 0)
+            if guesstilts is not None:
+                ytfit[wmask] -= np.median(ytfit[wmask])
 
-    # Plot the model tilts
-    tilts /= msarc.shape[0]
-    tilts += np.linspace(0.0, 1.0, msarc.shape[0]).reshape(-1, 1).repeat(msarc.shape[1], axis=1)
-    tiltsplot = tilts[arcdet, :].T
-    tiltsplot *= (msarc.shape[0]-1.0)
-    # ztilt[np.where(ztilt != maskval)] *= (msarc.shape[0]-1.0)
-    for i in xrange(arcdet.size):
-        w = np.where(ztilt[:, i] != maskval)
-        if guesstilts is not None: ztilt[w[0], i] += stilt[w[0], i]
-        if w[0].size != 0:
-            twa = (xtilt[w[0], i]+0.5).astype(np.int)
-            fitcns = np.polyfit(w[0], ztilt[w[0], i]-tiltsplot[twa, i], 0)[0]
-            if fitcns > 1.0:
-                msgs.warn("The tilt of Arc Line {0:d} might be poorly traced".format(i+1))
-            tiltsplot[:, i] += fitcns
-    xdat = xtilt.copy()
-    # xdat[np.where(xdat!=maskval)] *= (msarc.shape[1]-1.0)
-    if plotQA:
-        arplot.plot_orderfits(slf, tiltsplot, ztilt, xdata=xdat, xmodl=np.arange(msarc.shape[1]),
-                              textplt="Arc line", maxp=9, desc="Arc line spectral tilts", maskval=maskval)
-    return tilts, satsnd
+            # Perform a scanning polynomial fit to the tilts
+            model = arcyutils.polyfit_scan_intext(xtfit, ytfit, np.ones(ytfit.size, dtype=np.float), mtfit,
+                                                  2, sz/6, 3, maskval)
 
+            if maskval in model:
+                # Model contains masked values
+                aduse[j] = False
+                badlines += 1
+                continue
 
-def model_tilt(slf, det, msarc, guesstilts=None, censpec=None, plotQA=False, refine_tilts=False, maskval=-999999.9):
-    """
-    This function performs a PCA analysis on the arc tilts for a single spectrum (or order)
-    """
-    msgs.work("Detecting lines..")
-    tampl, tcent, twid, w, satsnd, _ = ararc.detect_lines(slf, det, msarc, censpec=censpec)
-    satval = slf._spect['det'][det-1]['saturation']*slf._spect['det'][det-1]['nonlinear']
-    fitxy = [slf._argflag['trace']['orders']['tiltorder'], 8]  # order of the polynomials to be used when fitting the tilts.
-    arcdet = (tcent[w]+0.5).astype(np.int)
-    maskrows = np.ones(msarc.shape[0], dtype=np.int) # Start by masking every row, then later unmask the rows with usable arc lines
-    totnum = 0
-    if np.size(w[0])>totnum:
-        totnum = np.size(w[0])
+            ytfit = (2.0*model[sz]-ytfit)
```

Comment:
Yes, but I've since realized that we get the (almost) right result for the wrong reasons. I'll fix this in the next PR.